File: src/main/java/dev/esophose/discordbot/webhook/WebhookExecuteRequest.java
package dev.esophose.discordbot.webhook;

import com.fasterxml.jackson.annotation.JsonProperty;
import discord4j.rest.json.request.EmbedRequest;
import java.util.ArrayList;
import java.util.List;
import reactor.util.annotation.Nullable;

public class WebhookExecuteRequest {

    @Nullable
    private final String content;
    @Nullable
    private final String username;
    @Nullable
    @JsonProperty("avatar_url")
    private final String avatarUrl;
    private final boolean tts;
    @Nullable
    private final List<EmbedRequest> embeds = new ArrayList<>();

    public WebhookExecuteRequest(@Nullable String content, @Nullable String username,
                                 @Nullable String avatarUrl, boolean tts, @Nullable EmbedRequest embed) {
        this.content = content;
        this.username = username;
        this.avatarUrl = avatarUrl;
        this.tts = tts;

        if (embed != null)
            this.embeds.add(embed);
    }

    @Override
    public String toString() {
        return "MessageCreateRequest{" +
                "content='" + this.content + '\'' +
                ", username='" + this.username + '\'' +
                ", avatar_url='" + this.avatarUrl + '\'' +
                ", tts=" + this.tts +
                ", embeds=" + this.embeds +
                '}';
    }
}

File: src/main/kotlin/dev/esophose/discordbot/webhook/WebhookExecuteRequest.kt
package dev.esophose.discordbot.webhook

import com.fasterxml.jackson.annotation.JsonProperty
import discord4j.discordjson.json.EmbedData
import reactor.util.annotation.Nullable
import java.util.ArrayList

class WebhookExecuteRequest(@param:Nullable @field:Nullable private val content: String?,
                            @param:Nullable @field:Nullable private val username: String?,
                            @param:Nullable @field:Nullable @field:JsonProperty("avatar_url") private val avatarUrl: String?,
                            private val tts: Boolean,
                            @Nullable embed: EmbedData?) {
    @Nullable
    private val embeds = ArrayList<EmbedData>()

    init {
        if (embed != null)
            this.embeds.add(embed)
    }

    override fun toString(): String {
        return "MessageCreateRequest{" +
                "content='" + this.content + '\'' +
                ", username='" + this.username + '\'' +
                ", avatar_url='" + this.avatarUrl + '\'' +
                ", tts=" + this.tts +
                ", embeds=" + this.embeds +
                '}'
    }
}
